README.md

```diff
@@ -1,105 +1,256 @@
+    ```
 
-### Test It Out
-1. Install the CRDs into the cluster:
+    > Note: this step is required only because the serverless-manager is in early stage and there is no manager for kyma CRDs installation
 
-```sh
-make install
-```
+4. Build and push Serverless Manager image
 
-2. Run your controller (this will run in the foreground, so switch to a new terminal if you want to leave it running):
+    ```bash
+    make module-image IMG_REGISTRY=localhost:5001/unsigned/operator-images IMG=localhost:5001/serverless-manager-dev-local:0.0.1
+    ```
 
-```sh
-make run
-```
+5. Build and push Serverless module
 
-**NOTE:** You can also run this in one step by running: `make install run`
+    ```bash
+    make module-build IMG=k3d-kyma-registry:5001/serverless-manager-dev-local:0.0.1 MODULE_REGISTRY=localhost:5001/unsigned
+    ```
 
-### Modifying the API definitions
-If you are editing the API definitions, generate the manifests such as CRs or CRDs using:
+6. Verify if the module and the manager's image are pushed to the local registry
 
-```sh
-make manifests
-```
+    ```bash
+    curl localhost:5001/v2/_catalog
+    ```
 
-**NOTE:** Run `make --help` for more information on all potential `make` targets
+    The example resoulte should looks like:
 
-More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
+    ```json
+    {"repositories":["serverless-manager-dev-local","unsigned/component-descriptors/kyma.project.io/module/serverless"]}
+    ```
 
-## License
+7. Patch CoreDNS resource on the Kyma cluster
 
-Copyright 2022.
+    ```bash
+    bash hack/get_kyma_localhost_registry_name.sh k3d-kyma-registry
+    ```
 
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
+8. Inpect the generated module template
 
-    http://www.apache.org/licenses/LICENSE-2.0
+    The following are temporary workarounds.
 
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
+    Edit the `template.yaml` file and:
 
+    - change `target` to `control-plane`
+
+    >**NOTE:** This is only required in the single cluster mode
+
+    ```yaml
+    spec:
+        target: control-plane
+    ```
+
+    - change the existing repository context in `spec.descriptor.component`:
+    >**NOTE:** Because Pods inside the k3d cluster use the docker-internal port of the registry, it tries to resolve the registry against port 5000 instead of 5001. K3d has registry aliases but module-manager is not part of k3d and thus does not know how to properly alias `k3d-kyma-registry.localhost:5001`
```

Comment:
```suggestion
    >**NOTE:** Because Pods inside the k3d cluster use the docker-internal port of the registry, it tries to resolve the registry against port 5000 instead of 5001. K3d has registry aliases, but `module-manager` is not part of k3d and thus does not know how to properly alias `k3d-kyma-registry.localhost:5001`
```